uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-requests_test.js

```diff
@@ -1,30 +1,899 @@
+                controller.filters.contact.location = mockLocation;
+                controller.refresh();
+              });
+
+              it('ContactAPI gets called with location value', function () {
+                expect(Contact.all).toHaveBeenCalledWith(jasmine.objectContaining({
+                  location: mockLocation.value
+                }), jasmine.any(Object));
+              });
+            });
+
+            describe('when location filter has no value', function () {
+
+              beforeEach(function () {
+                controller.filters.contact.location = null;
+                controller.refresh();
+              });
+
+              it('ContactAPI gets called with null', function () {
+                expect(Contact.all).toHaveBeenCalledWith(jasmine.objectContaining({
+                  location: null
+                }), jasmine.any(Object));
+              });
+            });
+          });
+
+          describe('level_type::', function () {
+
+            describe('when level_type filter is an empty array', function () {
+
+              beforeEach(function () {
+                controller.filters.contact.level_type = [];
+                controller.refresh();
+              });
+
+              it('ContactAPI gets called with null', function () {
+                expect(Contact.all).toHaveBeenCalledWith(jasmine.objectContaining({
+                  level_type: null
+                }), jasmine.any(Object));
+              });
+            });
+
+            describe('when level_type filter is not empty', function () {
+
+              var mockLevelType = [{
+                value: 'mockvalue'
+              }];
+
+              beforeEach(function () {
+                controller.filters.contact.level_type = mockLevelType;
+                controller.refresh();
+              });
+
+              it('ContactAPI gets called with level types', function () {
+                expect(Contact.all).toHaveBeenCalledWith(jasmine.objectContaining({
+                  level_type: {
+                    "IN": ['mockvalue']
+                  }
+                }), jasmine.any(Object));
+              });
+            });
+          });
+        });
+
+        describe('leaveRequestFilters', function () {
+
+          var promise,
+            defer;
+          beforeEach(function () {
+            defer = $q.defer();
+            spyOn(LeaveRequest, 'all').and.callThrough();
+            spyOn(controller, 'getUserNameByID');
+            Contact.all.and.returnValue(defer.promise);
+            promise = defer.promise;
+          });
+
+          afterEach(function () {
+            $rootScope.$apply();
+          });
+
+          describe('managed_by', function () {
+
+            beforeEach(function () {
+              controller.refresh();
+              defer.resolve({
+                list: []
+              });
+            });
+
+            it('filtered by managed_by', function () {
```

Comment:
Again please use more english-like, descriptive titles. Something like "it fetches only the leave requests managed by the user" makes it perfectly clear what's the expected behaviour of the component, and what you are testing